File: ObjectOrientedDesign(LibrarySearch)/Video.cpp
#include<iostream>
#include<string>
#include"Video.h"
#include"Media.h"
using namespace std;


Video :: Video( string a,string b,string c,string d, string e, string f, string g, string h) : Media(a, b, c, f)
{    
  this->description=d;
  this->distributor=e;
  this->series=g;
  this->label=h;  
}

void Video::display() const
{
  cout<<"<------------------------------ Record From Video--------------------------------------------->"<<endl;
  cout<<"Call Number:"<<call_number<<"\n"<<"Title:"<<title<<"\n"<<"Subjects:"<<subject<<"\n"<<"Description:"<<description<<
  "\n"<<"Distributor:"<<distributor<<"\n"<<"Notes:"<<notes<<"\n"<<"Series:"<<series<<"\n"<<"Label:"<<label<<endl;
}


bool Video::compare_other(const string& x) const
{  
  if (this->description.find(x) != std::string::npos)
  {
     return true;
  } 
  else if(this->distributor.find(x) !=std::string::npos) 
  {
     return true;
  }
   else if(this->notes.find(x) != std::string::npos)
  {
     return true;
  } 
  else
   {
     return false;
   }
}

File: Python_Library/Video.py
from Media import Media

class Video(Media):
	def __init__(self,a,b,c,d,e,f,g,h):
		Media.__init__(self,a,b,c,f)
		self.description=d
		self.distributor=e
		self.series=g
		self.label=h
		
		
	def display(self):
		print("<------------------------------ Record From Video --------------------------------------------->\n");
		print("Call Number: "+self.call_number+"\n");
		print("Title: "+self.title+"\n");
		print("Subjects: "+self.subject+"\n");
		print("Description: "+self.description+"\n");
		print("Distributor: "+self.distributor+"\n");
		print("Notes: "+self.notes+"\n");
		print("Series: "+self.series+"\n");
		print("Label: "+self.label+"\n");
		
	def compare_other(self,x):
		if (self.description.find(x) != -1):
			return True
		elif(self.distributor.find(x) !=-1):
			return True
		elif(self.notes.find(x) !=-1):
			return True
		else:
			return False
	
 
  
